src/common/components/Loader.tsx

```diff
@@ -0,0 +1,29 @@
+import { faSpinner } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+/**
+ * Component for rendering loading states
+ */
+export const Loader = (props: {
+  /**state of loader, defaults to true */
+  loading?: boolean;
+  /**style of loader to render when loading==true */
+  type?: 'text' | 'spinner';
+  /**Loader children are rendered when loading==false */
+  children?: React.ReactNode | React.ReactNode[];
+  /** Overrides the loader style to render a custom loader*/
+  render?: React.ReactNode | React.ReactNode[];
+}) => {
```

Comment:
I've got a stylistic question for you, but do not let it distract you too much.

Should we put more effort into a unified code style for functional components?

In abstract there is not much difference I can see between this version and one more like what I probably would have written:
```js
import { FC } from 'react';

interface LoaderProps {
  /**Loader children are rendered when loading==false */
  children?: React.ReactNode | React.ReactNode[];
  /**state of loader, defaults to true */
  loading?: boolean;
  /** Overrides the loader style to render a custom loader*/
  render?: React.ReactNode | React.ReactNode[];
  /**style of loader to render when loading==true */
  type?: 'text' | 'spinner';
}

export const Loader: FC<LoaderProps> = ({
  children,
  loading,
  render,
  type,
}) => {
  console.log({ children, loading, render, type });
  // etc...
};
```

In this case, I think there are not many differences between these approaches. The one that comes to my mind is that it is easier to export `LoaderProps` in case you need to reuse it (which is not _yet_ a concern for this component).

What do you think?